src/components/hint/index.tsx

```diff
@@ -436,25 +441,27 @@ class Hint extends Component<HintProps, HintState> {
   }
 
   renderHint() {
-    const {testID} = this.props;
-
+    const {onPress, testID} = this.props;
+    const opacity = onPress ? 0.9 : 1.0;
     if (this.showHint) {
       return (
-        <View
-          animated
-          style={[
-            {width: this.containerWidth},
-            styles.animatedContainer,
-            this.getHintPosition(),
-            this.getHintPadding(),
-            this.getHintAnimatedStyle()
-          ]}
-          pointerEvents="box-none"
-          testID={testID}
-        >
-          {this.renderHintTip()}
-          {this.renderContent()}
-        </View>
+        <TouchableOpacity activeOpacity={opacity} onPress={onPress}>
```

Comment:
Instead of wrapping with another component consider swapping between the wrapping container, we do it often in such cases.. 
For instance,
```
const Container = onPress ? TouchableOpacity : View;

<Container .....>
```